src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -457,10 +457,45 @@ bool TryLoadHostPolicy(StackSString& hostPolicyPath)
     {
         return true;
     }
+
+    // Use proper prefix / extension for the host policy dynamic library
+#if defined(__APPLE__)
+    static const WCHAR LibraryPrefix[] = u"lib";
+    static const WCHAR LibrarySuffix[] = u".dylib";
+    static const WCHAR PathSeparator = u'/';
+#elif defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__)
+    static const WCHAR LibraryPrefix[] = u"lib";
+    static const WCHAR LibrarySuffix[] = u".so";
+    static const WCHAR PathSeparator = u'/';
+#else
+    static const WCHAR LibraryPrefix[] = L"";
```

Comment:
It would be better to have libXXX.so as the fallback, and have the Windows-specific .dll under Windows-specific ifdef.

Reply:
Good point, fixed in 2nd commit.

Reply:
OK, as usual this turned out to be somewhat more involved than originally expected. After the Pri1 tests failed again, I found out to my dismay that we have two completely separate implementations of this part of CoreRun for Windows / Linux and I optimistically only fixed the Windows version where this is a no-issue. I have used this newly acquired knowledge to simplify the change substantially - I have given up on the attempt to "unify Windows and Linux library conventions" as it's no use for this change so I reverted my changes to the Windows version of corerun.cpp and to the appropriate section of the MockHosting script and I ported a much simplified version of the change to the Unix version, just choosing the appropriate extension for OSX vs. Linux. I'm about to send out the PR update shortly after initial local validation.

Reply:
@jkotas - I believe I have addressed your feedback and I have a passing CI and outerloop run. Could you please take another look when you have a chance?